File: project/ranyoung-onboarding/src/main/java/com/example/ranyoungonboarding/domain/Question.java
package com.example.ranyoungonboarding.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "questions")
@Getter
@Setter
public class Question {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id", nullable = false)
    private Survey survey;

    @Column(nullable = false)
    private String name;

    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private QuestionType type;

    @Column(nullable = false)
    private Boolean required;

    @ElementCollection
    @CollectionTable(name = "question_options", joinColumns = @JoinColumn(name = "question_id"))
    @Column(name = "option_value")
    private List<String> options = new ArrayList<>();

    @OneToMany(mappedBy = "question", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Answer> answers = new ArrayList<>();

    // position/order of the question in the survey
    private Integer position;
}

File: project/minjong-onboarding/src/main/kotlin/com/innercircle/survey/domain/Question.kt
// src/main/kotlin/com/innercircle/survey/domain/Question.kt
package com.innercircle.survey.domain

import javax.persistence.*

@Entity
@Table(name = "questions")
class Question(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(nullable = false)
    var questionName: String,

    @Column
    var questionDescription: String? = null,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var questionType: QuestionType,

    @Column(nullable = false)
    var required: Boolean = false,

    @Column(nullable = false)
    var deleted: Boolean = false,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "survey_id")
    var survey: Survey? = null,

    @OneToMany(mappedBy = "question", cascade = [CascadeType.ALL], orphanRemoval = true)
    val options: MutableList<Option> = mutableListOf(),

    @Column(nullable = false)
    var position: Int = 0
) {
    // 옵션 추가 메서드
    fun addOption(option: Option) {
        options.add(option)
        option.question = this
    }
}